From c206431859594032227ca93f66116a5838b3766b Mon Sep 17 00:00:00 2001
From: John Stebbins <jstebbins@jetheaddev.com>
Date: Wed, 11 Dec 2019 15:37:34 -0800
Subject: [PATCH] ccaption_dec: fix erase message pts in "real_time" mode

The erase message had the same timestamp as the last subtitle, thus
erasing it as soon as it was rendered.
---
 libavcodec/ccaption_dec.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavcodec/ccaption_dec.c b/libavcodec/ccaption_dec.c
index bf3563a0bc..afe290c5a6 100644
--- a/libavcodec/ccaption_dec.c
+++ b/libavcodec/ccaption_dec.c
@@ -700,6 +700,8 @@ static void process_cc608(CCaptionSubContext *ctx, int64_t pts, uint8_t hi, uint
             break;
         case 0x2c:
             /* erase display memory */
+            if (ctx->real_time)
+                ctx->startv_time = pts;
             handle_edm(ctx, pts);
             break;
         case 0x2d:
-- 
2.23.0

